Packs/FidelisEndpoint/Integrations/FidelisEndpoint/FidelisEndpoint.py

```diff
@@ -0,0 +1,1753 @@
+               'ParentID', 'EventType']
+
+    response = client.query_events(limit, start_time, end_time, logic, column, value, entity_type, operator,
+                                   additional_filter)
+    res = response.get('data', {})
+    events = res.get('events', [])
+    if not events:
+        return f'No events were found', {}, {}
+
+    for event in events:
+        contents.append({
+            'EndpointName': event.get('endpointName'),
+            'EventType': event.get('eventType'),
+            'ParentID': event.get('parentId'),
+            'PID': event.get('pid'),
+            'User': event.get('user'),
+            'ProcessStartTime': event.get('processStartTime'),
+            'LocalIP': event.get('localIP'),
+            'LocalPort': event.get('localPort'),
+            'RemoteIP': event.get('remoteIP'),
+            'RemotePort': event.get('remotePort')
+        })
+
+        context.append({
+            'EventTime': event.get('eventTime'),
+            'EndpointName': event.get('endpointName'),
+            'EventType': event.get('eventType'),
+            'ParentID': event.get('parentId'),
+            'TargetID': event.get('targetId'),
+            'PID': event.get('pid'),
+            'ParentName': event.get('parentName'),
+            'Name': event.get('name'),
+            'Path': event.get('path'),
+            'Hash': event.get('hash'),
+            'User': event.get('user'),
+            'LocalIP': event.get('localIP'),
+            'LocalPort': event.get('localPort'),
+            'RemoteIP': event.get('remoteIP'),
+            'RemotePort': event.get('remotePort'),
+            'DnsQuestion': event.get('dnsQuestion'),
+            'DnsAnswer': event.get('dnsAnswer'),
+            'ProcessStartTime': event.get('processStartTime'),
+            'EventIndex': event.get('eventIndex'),
+            'IndexingTime': event.get('indexingTime'),
+            'EntityType': event.get('entityType'),
+            'StartTime': event.get('startTime')
+        })
+
+    entry_context = {'FidelisEndpoint.Query(val.PID && val.PID === obj.PID)': context}
+    human_readable = tableToMarkdown('Fidelis Endpoint query events result', contents, headers=headers,
+                                     removeNull=True)
+    return human_readable, entry_context, response
+
+
+def parse_timestamp(time_):
+    if len(time_[time_.rfind('.')+1:time_.rfind('Z')]) > 3:
+        new_time = time_[:time_.rfind('.') + 4] + 'Z'
+    return new_time
+
+
+def fetch_incidents(client: Client, fetch_time: str, fetch_limit: str, last_run: Dict) -> Tuple[List, Dict]:
+    # handle first time fetch
+    last_run = last_run.get('time')
+    if not last_run:
+        last_run, _ = parse_date_range(fetch_time, to_timestamp=True)
+
+    current_fetch = last_run
+    incidents = []
+    last_fetch_date_string = timestamp_to_datestring(last_run, '%Y-%m-%dT%H:%M:%S.%fZ')
+    start_time = parse_timestamp(last_fetch_date_string)
```

Comment:
if i understand correctly, this function strips the milliseconds from the timestamp?
if so, why not just pass `%Y-%m-%dT%H:%M:%SZ` to be the date format?

Reply:
Not completely. It strirps the last milliseconds and leaves only 3 digits after the .
It changes the timestamp so the integration will know how to handle it. 
Basically instead of this: 2020-02-15T14:23:38.000000Z
We will get: 2020-02-15T14:23:38.000Z

Reply:
in this case just dont pass a date format and it will use the default `"%Y-%m-%dT%H:%M:%S.000Z"` and will return you the requested result